// This is used for convolution logging. Also see
// tensorflow/core/protobuf/autotuing.h
syntax = "proto3";

package xla.gpu;

import "tensorflow/compiler/xla/service/hlo.proto";
import "tensorflow/compiler/xla/xla_data.proto";
import "tensorflow/core/protobuf/autotuning.proto";

message ConvInstructionLog {
  xla.HloInstructionProto instruction = 1;
  repeated xla.ShapeProto operand_shapes = 2;
  uint64 result_address = 3;
  repeated uint64 operand_addresses = 4;
}

message CudnnConvAlgorithm {
  int64 id = 1;
  bool tensor_ops = 2;
}

message CudnnConvolutionEntry {
  string hlo = 1;
  tensorflow.ComputeCapability cc = 2;
  repeated tensorflow.CudnnVersion cudnn_versions = 3;
  repeated CudnnConvAlgorithm algos = 4;
}

message CudnnConvolutionList {
  repeated CudnnConvolutionEntry entries = 1;
}
